### PR TITLE
docs: Update analytics, canonical URL, ads

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

A bunch of changes to solidify the new docs site:

1. Correctly set the canonical URL on all pages
2. Set up Google Analytics
3. Externalized data into `en.yml` to match `new.eslint.org` and make it easier to change settings later
4. Updated carbon ad
5. Updated footer to use data from `en.yml`
6. Updated header to use data from `en.yml`

#### Is there anything you'd like reviewers to focus on?

Do the changes look correct?

### CODE REVIEW

File: docs/src/_includes/partials/docs-footer.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
             {% include 'components/social-icons.html' %}
         </div>
         <div class="docs-copyright">
-            &copy; OpenJS Foundation and other contributors, <a href="https://www.openjsf.org">www.openjsf.org</a>
+            <p>{{ site.footer.copyright | safe }}</p>
```
Comment (mdjermanovic) [inline docs/src/_includes/partials/docs-footer.html:7]: This `<p>` seems to add a bit of vertical space between the copyright line and the theme switcher. Is it intentional?

This PR's preview:

image

Current head version (https://new.eslint.org/docs/head/):

image

File: docs/src/_includes/layouts/base.html [hunk 2]
```diff
@@ -19,19 +31,19 @@
...
-    <meta property="og:title" content="{{ title }}">
-    <meta property="og:description" content="{{ desc }}">
-    <meta property="og:image" content="{{ cover }}" />
-    <meta property="og:image:alt" content="{{ coverAlt }}">
-    <meta property="og:locale" content="en_US">
+    <meta property="og:title" content="{{ page_title }}">
+    <meta property="og:description" content="{{ page_desc }}">
+    <meta property="og:image" content="{{ cover_image }}" />
+    <meta property="og:image:alt" content="{{ cover_image_alt }}">
+    <meta property="og:locale" content="{{ site.locale | replace(" -", "_" ) }}">
```
Comment (mdjermanovic) [inline docs/src/_includes/layouts/base.html:38]:
```suggestion
    <meta property="og:locale" content="{{ site.locale | replace("-", "_" ) }}">
```

There is an extra space, so this results in unchanged `en-US`

File: docs/src/_includes/partials/docs-footer.html (hunk 1 above)
Comment (nzakas) [reply docs/src/_includes/partials/docs-footer.html:7]:
```suggestion
            {{ site.footer.copyright | safe }}
```

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**